Rematch/UI.lua

```diff
@@ -9,251 +9,93 @@ local ns = select(2, ...)
-
-    local function move(button, add)
-        if not button:IsShown() then
-            return 0
-        end
+    -- When a script is added/removed, refresh the teams list.
+    self:RegisterMessage('PET_BATTLE_SCRIPT_SCRIPT_LIST_UPDATE')
+end
 
-        local point, relative, relativePoint, x, y = button:GetPoint()
-        button:SetPoint(point, relative, relativePoint, x + 21, y)
-        return add
+function Addon:PET_BATTLE_SCRIPT_SCRIPT_LIST_UPDATE()
+    if Rematch.teamsPanel:IsVisible() then
+        Rematch.teamsPanel:Refresh()
     end
-
-    self:SecureHook(RematchLoadedTeamPanel, 'Update', function(panel)
-        local footnotes = panel.Footnotes
-        local script = scriptButtons[footnotes]
-
-        if self:GetScript(RematchSettings.loadedTeam) then
-            script.key = RematchSettings.loadedTeam
-            script:Show()
-            script:ClearAllPoints()
-            script:SetPoint('LEFT', 5, -0.5)
-
-            local fx = 5 + 21
-            fx = fx + move(footnotes.Preferences, 21)
-            fx = fx + move(footnotes.Notes, 21)
-            fx = fx + move(footnotes.WinRecord, footnotes.WinRecord:GetWidth())
-            fx = fx + move(footnotes.Maximize, 21)
-            fx = fx + move(footnotes.Close, 21)
-
-            local footnoteWidth = fx + 4
-            local panelWidth = panel.maxWidth or 280
-
-            footnotes:SetWidth(footnoteWidth)
-            footnotes:Show()
-            panel:SetWidth(panelWidth-footnoteWidth-3)
-        else
-            script:Hide()
-        end
-    end)
-
-    self:HookScript(RematchJournal, 'OnShow', function(self)
-        CollectionsJournal:SetAttribute('UIPanelLayout-width', 870)
-        UpdateUIPanelPositions()
-    end)
-    self:HookScript(RematchJournal, 'OnHide', function(self)
-        CollectionsJournal:SetAttribute('UIPanelLayout-width', 710)
-        UpdateUIPanelPositions()
-    end)
 end
 
+
 function Addon:OnDisable()
-    tDeleteItem(Rematch:GetMenu('TeamMenu'), teamMenu)
-end
+    -- TODO: hide menu, disable badge?
 
-function Addon:FindMenuItem(menu, text)
-    for i, v in ipairs(menu) do
-        if v.text == text then
-            return v
-        end
-    end
+    -- Is there even a way to disable this plugin after its loaded? (outisde of source)
+
+    -- Should we Unregister REMATCH_TEAM_DELETED?
+    -- If a team is deleted when this plugin is disabled, matching scripts will
+    -- be made orphans, and if a new team is created with that same key, the
+    -- old script will be matched with it.
 end
 
 function Addon:OnExport(key)
-    if RematchSaved[key] then
-        Rematch:SetSideline(key, RematchSaved[key])
-        return Rematch:ConvertSidelineToString()
-    end
+    --if Rematch.savedTeams[key] then
+    --    return Rematch.teamStrings:ExportTeam(key)
+    --end
 end
 
 function Addon:OnImport(data)
-    Rematch:ShowImportDialog()
-    RematchDialog.MultiLine.EditBox:SetText(data)
+    -- TODO: Bug. (Do people even use this?)
```

Comment:
open todo

> TODO: Bug. (Do people even use this?)
> - When importing a PBS script, the key is preset in the data, but it wont match up after importing this Rematch team.

Reply:
I don't actually think people use this, but I kind of would love if people do. One of the more common requests on xufu is importing team and script at the same time, which would work with this. Thus, I'd prefer not to remove it.